reana_workflow_controller/rest/utils.py

```diff
@@ -625,20 +607,33 @@ def get_workspace_diff(workflow_a, workflow_b, brief=False, context_lines=5):
             )
 
 
-def get_workflow_progress(workflow, include_progress=False):
+def get_most_recent_job_info(workflow_id: UUID) -> Dict[str, str]:
+    """Return most recent Job cmd and name from a certain workflow."""
+    current_job_commands = {}
+    most_recent_job = (
+        Session.query(Job)
```

Comment:
nice improvement :+1:  one query instead of.. a lot :)

Reply:
The worse part was that this function was called always, independently of `verbose` or `include_progress` values 😬